### DISCUSSION TITLE
Batch mode test failure

### DISCUSSION THREAD

**Discussion by dkokron** [openucx/ucx#9868] [General]
I am attempting to use batch mode when running ucx_perftest.  The help page for this option has this to say.

     -b <file>      read and execute tests from a batch file: every line in the
                    file is a test to run, first word is test name, the rest of
                    the line is command-line arguments for the test.

alogin02 UCX > echo $p
9765

alogin02 UCX > cat tests.txt
get -d memory -D zcopy -x cma

alogin02 UCX > ucx-1.16.0/install/bin/ucx_perftest -c 1 -p $p &
[1] 4569
alogin02 UCX > Waiting for connection...

alogin02 UCX > ucx-1.16.0/install/bin/ucx_perftest alogin02 -c 2 -p $p -b tests.txt
Accepted connection from 10.253.0.46:49904
+--------------+--------------+------------------------------+---------------------+-----------------------+
|              |              |       overhead (usec)        |   bandwidth (MB/s)  |  message rate (msg/s) |
+--------------+--------------+----------+---------+---------+----------+----------+-----------+-----------+
|    Stage     | # iterations | 50.0%ile | average | overall |  average |  overall |  average  |  overall  |
+--------------+--------------+----------+---------+---------+----------+----------+-----------+-----------+
[1715368212.890024] [alogin02:4569 :0] perftest_params.c:444  UCX  ERROR missing test name
[1715368212.890195] [alogin02:4595 :0] perftest_params.c:444  UCX  ERROR in batch file 'tests.txt' line 1: missing test name
[1715368212.890222] [alogin02:4595 :0]    perftest_run.c:342  UCX  ERROR Failed to run test: Invalid parameter
[1]+  Exit 255                ucx-1.16.0/install/bin/ucx_perftest -c 1 -p $p

Can anyone tell me what I'm doing wrong?

**Comment by yosefe** [comment]
@dkokron pls see https://github.com/openucx/ucx/blob/4e10990960ca4a92d94e713ca8c155016ae51cee/contrib/ucx_perftest_config/test_types_uct#L18-L19 as example of a batch file running in CI

**Comment by dkokron** [comment]
Thank you!

**Comment by dkokron** [comment]
If the first element on each line is supposed to be a user defined test name, please consider changing the text description for the "-b" option to reflect that.
"first word is a user defined test name"

**Reply by michal-shalev** [reply]
Hi @dkokron, 
Thank you for pointing that out. We have updated the help message to reflect that the first word is a user-defined test name. You can see the fix here: https://github.com/openucx/ucx/pull/9916
